destination/destination_integration_test.go

```diff
@@ -0,0 +1,395 @@
+	testDB               = "test"
+	testCollectionPrefix = "test_coll"
+
+	// next consts will be used for test models.
+	testIDFieldName = "_id"
+	testID          = "6384c3e48740eb54f858bde5"
+
+	testTextFieldName = "textfield"
+	testTextData      = "some text data"
+
+	testNumberFieldName  = "numberfield"
+	testNumberData       = 1234
+	testNumberDataChange = 5678
+)
+
+func TestDestination_Write_snapshotSuccess(t *testing.T) {
+	is := is.New(t)
+
+	cfg := prepareConfig(t)
+
+	destination := NewDestination()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.NoErr(err)
+
+	col, err := getTestCollection(ctx, cfg[config.KeyCollection])
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = col.Drop(context.Background())
+		is.NoErr(err)
+
+		err = destination.Teardown(ctx)
+		is.NoErr(err)
+	})
+
+	n, err := destination.Write(ctx, []sdk.Record{getTestSnapshot(t)})
+	is.NoErr(err)
+	is.Equal(n, 1)
+
+	c, err := col.CountDocuments(ctx, bson.D{})
+	is.NoErr(err)
+	is.Equal(c, int64(1))
+
+	res, err := col.Find(ctx, bson.M{})
+	res.Next(ctx)
+	var result map[string]any
+	err = res.Decode(&result)
+	is.NoErr(err)
+
+	id, ok := result[testIDFieldName].(primitive.ObjectID)
+	is.Equal(ok, true)
+
+	hex, err := primitive.ObjectIDFromHex(testID)
+	is.NoErr(err)
+
+	is.Equal(id, hex)
+
+	text, ok := result[testTextFieldName].(string)
+	is.Equal(ok, true)
+	is.Equal(text, testTextData)
+
+	num, ok := result[testNumberFieldName].(float64)
+	is.Equal(ok, true)
+	is.Equal(int(num), testNumberData)
+
+	_, err = col.DeleteMany(ctx, bson.M{})
+	is.NoErr(err)
+}
+
+func TestDestination_Write_insertSuccess(t *testing.T) {
+	is := is.New(t)
+
+	cfg := prepareConfig(t)
+
+	destination := NewDestination()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := destination.Configure(ctx, cfg)
+	is.NoErr(err)
+
+	err = destination.Open(ctx)
+	is.NoErr(err)
+
+	col, err := getTestCollection(ctx, cfg[config.KeyCollection])
+	is.NoErr(err)
+
+	t.Cleanup(func() {
+		err = col.Drop(context.Background())
+		is.NoErr(err)
+
+		err = destination.Teardown(ctx)
+		is.NoErr(err)
+	})
+
+	n, err := destination.Write(ctx, []sdk.Record{getTestCreateRecord(t)})
+	is.NoErr(err)
+	is.Equal(n, 1)
+
+	c, err := col.CountDocuments(ctx, bson.D{})
+	is.NoErr(err)
+	is.Equal(c, int64(1))
+
+	res, err := col.Find(ctx, bson.M{})
+	res.Next(ctx)
+	var result map[string]any
+	err = res.Decode(&result)
+	is.NoErr(err)
+
+	id, ok := result[testIDFieldName].(primitive.ObjectID)
+	is.Equal(ok, true)
+
+	hex, err := primitive.ObjectIDFromHex(testID)
+	is.NoErr(err)
+
+	is.Equal(id, hex)
+
+	text, ok := result[testTextFieldName].(string)
+	is.Equal(ok, true)
+	is.Equal(text, testTextData)
+
+	num, ok := result[testNumberFieldName].(float64)
+	is.Equal(ok, true)
+	is.Equal(int(num), testNumberData)
```

Comment:
Since we have a similar code in all the test functions, maybe we can provide a single function with these checks?

Reply:
fixed